app/src/test/java/com/example/userinfo/MainActivityTest.kt

```diff
@@ -0,0 +1,62 @@
+package com.example.userinfo
+
+import android.widget.EditText
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import io.mockk.impl.annotations.MockK
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Rule
+
+import org.junit.Test
+import org.junit.rules.TestRule
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+class MainActivityTest {
+
+    @MockK
+    private lateinit var mainActivity: MainActivity
+
+    @get:Rule
+    var rule: TestRule = InstantTaskExecutorRule()
+
+    @Before
+    fun setUp() {
+        mainActivity = MainActivity()
+    }
+
+    @Test
+    fun isValidEmail() {
+        assertTrue(mainActivity.isValidEmail("bhavana@gmail.com"))
```

Comment:
If we can extract data validation into another class. We can write Local tests only. we do not need activity for this